CONTRIBUTING.md

```diff
@@ -0,0 +1,122 @@
+# Contributing
+
+The development flow includes:
+
+- [create workspace, clone sources](#create-workspace-clone-sources)
+- [build](#build)
+- [run tests](#run-tests)
+- [package binaries](#package-binaries)
+- [code](#code)
+- [push changes, provide pull request](#push-changes-provide-pull-request)
+
+## Create workspace, clone sources
+
+To create a workpace you can use [devfile](devfile.yaml).
+For that you have to [install the latest version](https://www.eclipse.org/che/docs/che-7/che-quick-starts.html#installing-the-chectl-management-tool) of this chectl and create a workspace:
+
+```
+$ chectl workspace:start -f https://raw.githubusercontent.com/che-incubator/chectl/master/devfile.yaml
+```
+
+> See more about [devfile](https://redhat-developer.github.io/devfile)
+
+After starting the workspace Theia will clone sources of `chectl` to `/projects/chectl` directory. Sources will be accessible in `theia-dev` and `dev` containers.
+
+#### 'theia-dev' container
+
+This container is used only for running Theia editor and does not take part in the following flow.
+
+#### 'dev' container
+
+`chectl` is written on TypeScript. For that this container has preinstalled software to be able to build, test and launch `chectl`.
+
+You workspace is initialized with a list of commands described in the [devfile](devfile.yaml) in `commands` section. Those command allow you to:
+- build
+- test
+- run
+- install packager
+- package binaries
+- format sources
+
+You can run commands in one of tree ways.
+
+1. It an easiest way is to use `My Workspace` panel at the left. You can launch commands by simple click.
+
+2. `Terminal => Run Task...` menu. You can fin and pick a command in the appeared command palette.
+
+3. Manually by opening a terminal in the `dev` container and running commands in `/projects/chectl` directory.
+
+## Build
+
+```bash
+yarn
+```
+
+Open `My Workspace` panel at the left and launch `Build` command. It will run `yarn` command in `/projects/chectl` directory inside `dev` container. The command will install all necessary dependencies and compile the project. Upon successfull assembly, a new `bin` directory will appear in project directory and will have two files: `run` and `run.cmd`.
+
+
+## Run tests
+
+```bash
+yarn test
+```
+
+Tests for `chectl` are written using `jest` framework.
+To run tests, go to `My Workspace` panel, find and launch `Test` command. The command will run `yarn test` inside `dev` container.
+Testing summary will be printed to the output.
+
+
+## Run
+
+```bash
+./bin/run --help
+```
+
+To test ensure `chectl` is built successfully, launch `Run` command. It wil run `chectl` with `--help` directive.
+
+## Package binaries
+
+### Install packager
```

Comment:
needs to be reworked. It's only using now `oclif-dev pack`

Reply:
Reworked